File: backend/src/admin/objects/apiTypes/AdminOrderMonitorResponse.scala
package delivery.admin.objects.apiTypes

import delivery.order.objects.Order

final case class AdminOrderMonitorItem(
    order: Order,
    reason: String,
    elapsedMinutes: Int
)

final case class AdminOrderMonitorResponse(
    todayOrderCount: Int,
    todayTurnover: Double,
    pendingRefunds: List[AdminOrderMonitorItem],
    abnormalOrders: List[AdminOrderMonitorItem],
    merchantTimeoutOrders: List[AdminOrderMonitorItem],
    riderTimeoutOrders: List[AdminOrderMonitorItem]
)

File: frontend/src/objects/admin/apiTypes/AdminOrderMonitorResponse.ts
import type { Order } from '@/objects/order/Order'

export interface AdminOrderMonitorItem {
  order: Order
  reason: string
  elapsedMinutes: number
}

export interface AdminOrderMonitorResponse {
  todayOrderCount: number
  todayTurnover: number
  pendingRefunds: AdminOrderMonitorItem[]
  abnormalOrders: AdminOrderMonitorItem[]
  merchantTimeoutOrders: AdminOrderMonitorItem[]
  riderTimeoutOrders: AdminOrderMonitorItem[]
}
